internal/controller/management/projectconfigs/project_configs.go

```diff
@@ -0,0 +1,311 @@
+		"webhook-receiver-configs", len(pc.Spec.WebhookReceivers),
+	)
+
+	var errs []error
+	webhookReceivers := make([]kargoapi.WebhookReceiver, 0, len(pc.Spec.WebhookReceivers))
+	for _, rc := range pc.Spec.WebhookReceivers {
+		whr, err := r.newWebhookReceiver(ctx, pc, rc)
+		if err != nil {
+			logger.Error(err, "error initializing new webhook receiver",
+				"receiver-config", rc,
+			)
+			errs = append(errs, fmt.Errorf(
+				"error initializing webhook receiver %q: %w",
+				rc.Name, err,
+			),
+			)
+			continue
+		}
+		webhookReceivers = append(webhookReceivers, *whr)
+	}
+	return webhookReceivers, kerrors.Flatten(kerrors.NewAggregate(errs))
+}
+
+func (r *reconciler) newWebhookReceiver(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+	rc kargoapi.WebhookReceiverConfig,
+) (*kargoapi.WebhookReceiver, error) {
+	logger := logging.LoggerFromContext(ctx)
+	cfg, err := getProviderConfig(rc)
+	if err != nil {
+		logger.Error(err, "error getting provider config",
+			"receiver-config", rc,
+			"project-config", pc,
+		)
+		return nil, fmt.Errorf("error getting secret name: %w", err)
+	}
+
+	var s corev1.Secret
+	err = r.client.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: pc.Namespace,
+			Name:      cfg.secretName,
+		},
+		&s,
+	)
+	if err != nil {
+		logger.Error(err, "secret not found",
+			"secret-name", cfg.secretName,
+		)
+		if kubeerr.IsNotFound(err) {
+			return nil, fmt.Errorf(
+				"secret %q in namespace %q not found",
+				cfg.secretName, pc.Namespace,
+			)
+		}
+		return nil, fmt.Errorf(
+			"error getting webhook receiver secret %q in project namespace %q: %w",
+			cfg.secretName, pc.Namespace, err,
+		)
+	}
+	logger.Debug("secret found", "secret", cfg.secretName)
+
+	secret, ok := s.Data[cfg.targetKey]
+	if !ok {
+		logger.Error(err, "target key not found in secret data",
+			"target-key", cfg.targetKey,
+		)
+		return nil, fmt.Errorf(
+			"key %q not found in secret %q for project config %q",
+			cfg.targetKey, cfg.secretName, pc.Name,
+		)
+	}
+
+	wr := &kargoapi.WebhookReceiver{
+		Name: rc.Name,
+		Path: external.GenerateWebhookPath(
+			pc.Name,
+			cfg.receiverType,
+			string(secret),
+		),
+	}
+	wr.URL = path.Join(r.cfg.ExternalWebhookServerBaseURL, wr.Path)
+	logger.Debug("webhook receiver initialized",
+		"webhook-receiver", wr,
+	)
+	return wr, nil
+}
+
+type providerConfig struct {
+	secretName   string
+	targetKey    string
+	receiverType string
+}
+
+func getProviderConfig(rc kargoapi.WebhookReceiverConfig) (*providerConfig, error) {
+	switch {
+	case rc.GitHub != nil:
+		if rc.GitHub.SecretRef.Name == "" {
+			return nil, errors.New("receiver config does not have a secret reference name")
+		}
+		return &providerConfig{
+			secretName:   rc.GitHub.SecretRef.Name,
+			targetKey:    kargoapi.WebhookReceiverSecretKeyGithub,
+			receiverType: kargoapi.WebhookReceiverTypeGitHub,
+		}, nil
+	default:
+		return nil, errors.New("webhook receiver config does not have any valid configs")
```

Comment:
This seems to redundant. Maybe:

```suggestion
		return nil, errors.New("webhook receiver config has no valid entry")
```